scipy/optimize/_linesearch.py

```diff
@@ -322,10 +322,10 @@ def extra_condition2(alpha, phi):
     return alpha_star, fc[0], gc[0], phi_star, old_fval, derphi_star
 
 
-def scalar_search_wolfe2(phi, derphi, phi0=None,
+def scalar_search_wolfe2(phi, derphi, maxiter = 10, phi0=None,
                          old_phi0=None, derphi0=None,
                          c1=1e-4, c2=0.9, amax=None,
-                         extra_condition=None, maxiter=10):
+                         extra_condition=None):
```

Comment:
```suggestion
def scalar_search_wolfe2(phi, derphi, phi0=None,
                         old_phi0=None, derphi0=None,
                         c1=1e-4, c2=0.9, amax=None,
                         extra_condition=None, maxiter=10):
```
Changing the location of the `maxiter` parameter can break backwards compatibility, so it should be left as-is.